commands/fetch_templates.go

```diff
@@ -4,263 +4,171 @@
+	}
 
-		switch action {
+TEMPLATES:
+	for _, file := range templates {
+		if !file.IsDir() {
+			continue TEMPLATES
+		}
+		language := file.Name()
 
-		case shouldExtractData:
-			expandFromZip = true
-		case newTemplateFound:
-			expandFromZip = true
+		canWrite := canWriteLanguage(availableLanguages, language, overwrite)
+		if canWrite {
 			fetchedLanguages = append(fetchedLanguages, language)
-		case directoryAlreadyExists:
-			expandFromZip = false
+			// Do cp here
+			languageSrc := filepath.Join(templateDir, language)
+			languageDest := filepath.Join(templateDirectory, language)
+			copy(languageSrc, languageDest, file)
+		} else {
 			existingLanguages = append(existingLanguages, language)
-		case skipWritingData:
-			expandFromZip = false
-		default:
-			return nil, nil, fmt.Errorf(fmt.Sprintf("don't know what to do when extracting zip: %s", archivePath))
-		}
-
-		if expandFromZip {
-			var rc io.ReadCloser
-
-			if rc, err = z.Open(); err != nil {
-				break
-			}
-			defer rc.Close()
-
-			if err = createPath(relativePath, z.Mode()); err != nil {
-				break
-			}
-
-			// If relativePath is just a directory, then skip expanding it.
-			if len(relativePath) > 1 && !isDirectory {
-				if err = writeFile(rc, z.UncompressedSize64, relativePath, z.Mode()); err != nil {
-					return nil, nil, err
-				}
-			}
+			continue TEMPLATES
 		}
 	}
 
 	return existingLanguages, fetchedLanguages, nil
 }
 
-// canExpandTemplateData returns what we should do with the file in form of ExtractAction enum
-// with the language name and whether it is a directory
-func canExpandTemplateData(availableLanguages map[string]bool, relativePath string) (extractAction, string, bool) {
-	if pathSplit := strings.Split(relativePath, "/"); len(pathSplit) > 2 {
-		language := pathSplit[1]
-
-		// We know that this path is a directory if the last character is a "/"
-		isDirectory := strings.HasSuffix(relativePath, "/")
-
-		// Check if this is the root directory for a language (at ./template/lang)
-		if len(pathSplit) == rootLanguageDirSplitCount && isDirectory {
-			if !canWriteLanguage(availableLanguages, language, overwrite) {
-				return directoryAlreadyExists, language, isDirectory
-			}
-			return newTemplateFound, language, isDirectory
-		}
-
-		if canWriteLanguage(availableLanguages, language, overwrite) == false {
-			return skipWritingData, language, isDirectory
-		}
-
-		return shouldExtractData, language, isDirectory
+// "recursively copy a file object, info must be non-nil
+func copy(src, dest string, info os.FileInfo) error {
+	if info.IsDir() {
+		return dcopy(src, dest, info)
 	}
-	// template/
-	return skipWritingData, "", true
+	return fcopy(src, dest, info)
 }
 
-// removeArchive removes the given file
-func removeArchive(archive string) error {
-	log.Printf("Cleaning up zip file...")
-	var err error
+// fcopy will copy a file with the same mode as the src file
+func fcopy(src, dest string, info os.FileInfo) error {
```

Comment:
What was this called before? Why do we need it? Does it also exist in the "build.go" code?

Reply:
I created this new, so it wasn't anything before. `copy`, `fcopy`, and `dcopy` are each new in this PR.  I included it before I realized there was a copy in the `build.go`.  I then tried to convert to that version of `Copy` but it failed, it was copying with an unexpected prefix and just generally didn't work as expected.  I wasn't sure what was not working there and I didn't want to break the build code, so I went back to this version with the new `copy` command.